### ISSUE TITLE
A false positive about the rule DM_EXIT

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#1763]
**SpotBugs Version:** 4.4.2
**Rule:** [DM_EXIT](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#dm-method-invokes-system-exit-dm-exit)

Hi! I find a false positive about the rule DM_EXIT. SpotBugs should not report a warning at line 7. We need to consider the initialization of final variable.

**Minimal Code Sample**
```java
public static final boolean tag;
static {
    tag = false;
}
public void func() {
    if(tag) {
        System.exit(0);  // A false warning reported here
    }
}
```
**Expected outcome:**
SpotBugs shouldn't report a violation at line 7, but it does. This is a false positive.

**Running SpotBugs through:** [CLI]
